distributed/recreate_exceptions.py

```diff
@@ -0,0 +1,98 @@
+from __future__ import print_function, division, absolute_import
+
+import logging
+from tornado import gen
+from .client import futures_of
+from.worker import _deserialize
+
+logger = logging.getLogger(__name__)
+
+
+class ExceptionScheduler(object):
```

Comment:
"Exceptions" may be a bit too broad of a term to claim for this functionality.  Dask handles exceptions in a variety of ways.